src/Pos/Wallet/KeyStorage.hs

```diff
@@ -194,9 +194,7 @@ instance (MonadIO m, MonadThrow m) =>
     addSecretKey sk =
         whenM (not . elem sk <$> use usKeys) $
             usKeys <>= [sk]
-    deleteSecretKey (fromIntegral -> i)
-        | i == 0 = throwM $ PrimaryKey "Cannot delete a primary secret key"
-        | otherwise = usKeys %= deleteAt i
+    deleteSecretKey (fromIntegral -> i) = usKeys %= deleteAt i
```

Comment:
I'm afraid this won't do. These changes were introduced after `UserSecret` format was changed and primary key started to be stored separately.